src/components/Facility/FacilityForm.tsx

```diff
@@ -118,19 +127,75 @@ export default function CreateFacilityForm({
     },
   });
 
-  const onSubmit = (data: FacilityFormValues) => {
-    createFacility({
+  const { mutate: updateFacility, isPending: isUpdatePending } = useMutation({
+    mutationFn: mutate(routes.updateFacility, {
+      pathParams: { id: facilityId || "" },
+    }),
+    onSuccess: (_data: FacilityModel) => {
+      toast.success(t("facility_updated_successfully"));
+      queryClient.invalidateQueries({ queryKey: ["organizationFacilities"] });
+      form.reset();
+      onSubmitSuccess?.();
+    },
+    onError: (error: Error) => {
+      const errorData = error.cause as { errors: { msg: string[] } };
+      if (errorData?.errors?.msg) {
+        errorData.errors.msg.forEach((msg) => {
+          toast.error(msg);
+        });
+      } else {
+        toast.error(t("facility_update_error"));
+      }
+    },
+  });
+
+  const { data: facilityData } = useQuery({
+    queryKey: ["facility", facilityId],
+    queryFn: query(routes.getPermittedFacility, {
+      pathParams: { id: facilityId || "" },
+    }),
+    enabled: !!facilityId,
+  });
+
+  const onSubmit: (data: FacilityFormValues) => void = (
+    data: FacilityFormValues,
+  ) => {
+    const requestData = {
       ...data,
       phone_number: parsePhoneNumber(data.phone_number),
-      geo_organization: organizationId,
-    });
+    };
+
+    if (facilityId) {
+      updateFacility(requestData);
+    } else {
+      createFacility(requestData);
+    }
   };
 
   const handleFeatureChange = (value: any) => {
     const { value: features }: { value: Array<number> } = value;
     form.setValue("features", features);
   };
 
+  // Update form when facility data is loaded
+  useEffect(() => {
+    if (facilityData) {
+      form.reset({
+        facility_type: facilityData.facility_type,
+        name: facilityData.name,
+        description: facilityData.description || "",
+        features: facilityData.features || [],
+        pincode: facilityData.pincode?.toString() || "",
+        geo_organization: facilityData.geo_organization || "",
```

Comment:
geo_organization here would be an object rather than a string.

Update return type for `getPermittedFacility` to be `FacilityData` rather than FacilityModel (make any additional changes as needed when updating this route).